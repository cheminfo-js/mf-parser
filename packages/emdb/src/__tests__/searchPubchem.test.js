'use strict';

const searchPubchem = require('../searchPubchem.js');

test('test searchPubchem', async () => {
  let data = await searchPubchem(60, {
    ionizations: '(H+)2, H+',
    precision: 10
  });
  expect(data).toMatchSnapshot();
});
