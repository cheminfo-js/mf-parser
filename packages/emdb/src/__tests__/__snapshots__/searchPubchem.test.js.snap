// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchPubchem simple case 1`] = `
Array [
  Object {
    "atoms": Object {
      "C": 2,
      "H": 4,
      "N": 2,
      "P": 2,
    },
    "charge": 0,
    "em": 117.98497213462,
    "info": Object {
      "nbPubchemEntries": 7,
    },
    "ionization": Object {
      "charge": 2,
      "em": 2.01565006446,
      "mf": "(H+)2",
    },
    "mass": 118.0141652294252,
    "mf": "C2H4N2P2",
    "monoisotopicMass": 117.98497213462,
    "ms": Object {
      "charge": 2,
      "delta": 0.00023748036907278447,
      "em": 59.99976251963093,
      "ionization": "(H+)2",
      "ppm": 3.958021817087773,
    },
    "unsaturation": 3,
  },
  Object {
    "atoms": Object {
      "C": 1,
      "H": 2,
      "N": 1,
      "P": 1,
    },
    "charge": 0,
    "em": 58.99248606731,
    "info": Object {
      "nbPubchemEntries": 9,
    },
    "ionization": Object {
      "charge": 1,
      "em": 1.00782503223,
      "mf": "H+",
    },
    "mass": 59.0070826147126,
    "mf": "CH2NP",
    "monoisotopicMass": 58.99248606731,
    "ms": Object {
      "charge": 1,
      "delta": 0.00023748036907278447,
      "em": 59.99976251963093,
      "ionization": "H+",
      "ppm": 3.958021817087773,
    },
    "unsaturation": 2,
  },
]
`;
